opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -218,11 +261,24 @@ public SpanBuilder withStartTimestamp(long microseconds) {
 
         @Override
         public MockSpan start() {
+            return startManual();
+        }
+
+        @Override
+        public ActiveSpan startActive() {
+            MockSpan span = this.startManual();
+            return spanSource.makeActive(span);
+        }
+
+        @Override
+        public MockSpan startManual() {
             if (this.startMicros == 0) {
                 this.startMicros = MockSpan.nowMicros();
             }
-            return new MockSpan(MockTracer.this, this.operationName, this.startMicros, initialTags, this.firstParent);
+            if (firstParent == null && !ignoringActiveSpan) {
+                firstParent = (MockSpan.MockContext)spanSource.activeSpan().context();
```

Comment:
`firstParent` is set at SpanBuilder creation time and also here. We should say _exactly_ when an activeParent is going to be inferred.